pkg/quota/v1/install/update_filter.go

```diff
@@ -44,3 +51,17 @@ func DefaultUpdateFilter() func(resource schema.GroupVersionResource, oldObj, ne
 		return false
 	}
 }
+
+// hasAllocatedResourcesChanged function to compare allocated resources in container statuses
+func hasAllocatedResourcesChanged(oldPod *v1.Pod, newPod *v1.Pod) bool {
+	if len(oldPod.Status.ContainerStatuses) != len(newPod.Status.ContainerStatuses) {
+		return false
+	}
+	for i, oldStatus := range oldPod.Status.ContainerStatuses {
+		newStatus := newPod.Status.ContainerStatuses[i]
+		if !resourcehelper.EqualResourceList(oldStatus.AllocatedResources, newStatus.AllocatedResources) {
```

Comment:
Would `maps.Equal` work here?
```suggestion
		if !maps.Equal(oldStatus.AllocateResources, newStatus.AllocateResources) { 
```

Reply:
This won't work because maps.Equal doesn't handle equal semantic value. 
otherwise, I will modify the "Equal ResourceList" test case to test this scenario. 
( CPU: 1000m  == CPU:1 )